client/src/main/java/io/confluent/kafka/schemaregistry/utils/QualifiedSubject.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafka.schemaregistry.utils;
+
+import java.util.Objects;
+
+public class QualifiedSubject implements Comparable<QualifiedSubject> {
+
+  public static final String DEFAULT_TENANT = "default";
+
+  public static final String DEFAULT_CONTEXT = ".";
+
+  public static final String TENANT_DELIMITER = "_";
+
+  public static final String CONTEXT_DELIMITER = ":";
+
+  public static final String CONTEXT_SEPARATOR = ".";
+
+  public static final String CONTEXT_PREFIX = CONTEXT_DELIMITER + CONTEXT_SEPARATOR;
+
+  private static final String WILDCARD = "*";
+
+  private final String tenant;
+  private final String context;
+  private final String subject;
+
+  // visible for testing
+  protected QualifiedSubject(String tenant, String qualifiedSubject) {
+    String contextSubject = qualifiedSubject != null ? qualifiedSubject : "";
+    if (!DEFAULT_TENANT.equals(tenant)) {
+      int ix = contextSubject.indexOf(TENANT_DELIMITER);
+      if (ix >= 0) {
+        String tenantPrefix = contextSubject.substring(0, ix);
+        contextSubject = contextSubject.substring(ix + 1);
+        // set actual tenant, we only really care whether the passed in tenant was "default"
+        tenant = tenantPrefix;
```

Comment:
Is it likely that _tenant_ parameter is non-default, but it's different from tenantPrefix? Should we enforce them to be the same?

Reply:
The qualifiedSubject cannot determine the tenant if it is default.  The other option would be to pass in a flag (instead of a tenant) whether the tenant is the default or not.